Xamarin.Forms.Core/AbsoluteLayout.cs

```diff
@@ -250,25 +250,19 @@ static Rectangle ComputeLayoutForRegion(View view, Size region)
 
 			if (!widthIsProportional && bounds.Width == AutoSize)
 			{
+				sizeRequest = view.Measure(region.Width, region.Height, MeasureFlags.IncludeMargins);
+				result.Width = sizeRequest.Request.Width > 0 ? sizeRequest.Request.Width : region.Width;
```

Comment:
I don't think this is quite right and that 0 means AutoSize. This change means if I set the WidthRequest on anything to zero it becomes full width

For example
```C#
Content = new AbsoluteLayout()
			{
				Children = {
					new Label(){ Text = "cat", WidthRequest = 0},
					new Image(){ Source = "coffee.png", WidthRequest = 0},
					new ActivityIndicator() { IsRunning = true }
				}
			};
```

Causes that Label and Image to be full width

If you change out AbsoluteLayout for a StackLayout you'll see that StackLayout will measure the Label and Image to have zero width because a WidthRequest of zero means please only make the width zero.


The fix for this Issue feels like it should be inside the ActivityIndcatorRenderer on UWP
If there's a way there to see that you are allowed to autosize to just fill the region you are in

Reply:
💯